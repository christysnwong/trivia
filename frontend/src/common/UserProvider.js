import React, {useEffect, useState} from "react";
import UserContext from "./UserContext";

const demoUser = {
  id: 1,
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "test@test.com",
  stats: {
    level: 14,
    title: "Pro",
    "quizzes completed": 34,
    points: 3440,
    remainingPts: 60,
    levelPts: 300,
  },
};

const UserProvider = ({
  children,
  currUser = demoUser
}) => {
  
  const [currentUser, setCurrUser] = useState();

  return (
    // <UserContext.Provider value={{ currUser }}>
    <UserContext.Provider value={{ currUser, setCurrUser }}>
      {children}
    </UserContext.Provider>
  );
};

export default UserProvider;
