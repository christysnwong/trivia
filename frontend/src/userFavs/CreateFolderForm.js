import { useContext, useState } from "react";
import UserContext from "../common/UserContext";
import JoblyApi from "../api/api";

import Noty from 'noty';

const CreateFolderForm = ({
  setUserFolders,
  setCreateFormIsOpen,
}) => {
  const { currUser } = useContext(UserContext);

  const [formData, setFormData] = useState({ folderName: "" });

  // handle changes in the form
  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((formData) => ({
      ...formData,
      userId: currUser.id,
      [name]: value,
    }));
  };

  // handle form submission and reset form
  const handleSubmit = async (evt) => {
    evt.preventDefault();

    try {
      let res = await JoblyApi.createFolder(currUser.username, formData);

      if (res.created) {

        new Noty({
          type: "success",
          layout: "topRight",
          theme: "bootstrap-v4",
          text: `Your folder ${formData.folderName} is successfully created`,
          timeout: "3000",
          progressBar: true,
          closeWith: ["click"],
          killer: true,
        }).show();


        setUserFolders((folder) => [
          ...folder,
          { folderId: res.created.folderId, name: res.created.name },
        ]);
        setCreateFormIsOpen(false);
      }

      setFormData({ folderName: "" });
    } catch (e) {
      console.error(e);

      new Noty({
        type: "error",
        layout: "topRight",
        theme: "bootstrap-v4",
        text: e,
        timeout: "3000",
        progressBar: true,
        closeWith: ["click"],
        killer: true,
      }).show();
    }
  };

  return (
    <>
      <div className="card mb-5">
        <div className="card-body">
          <form onSubmit={handleSubmit}>
            <div className="mb-2">
              <label htmlFor="username" className="form-label">
                Create a Folder:{" "}
              </label>

              <input
                id="folderName"
                name="folderName"
                className="form-control"
                value={formData.folderName}
                onChange={handleChange}
                placeholder="New Folder Name"
                maxLength="20"
              />
            </div>

            <button
              type="button"
              className="btn btn-secondary mt-4 me-3"
              onClick={() => setCreateFormIsOpen(false)}
              method="get"
            >
              Cancel
            </button>

            <button
              type="submit"
              className="btn btn-primary mt-4"
              method="post"
            >
              Create
            </button>
          </form>
        </div>
      </div>
    </>
  );
};

export default CreateFolderForm;