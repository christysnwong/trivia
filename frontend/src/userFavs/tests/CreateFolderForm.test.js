import React from "react";
import { render } from "@testing-library/react";
import CreateFolderForm from "../CreateFolderForm";
import { MemoryRouter } from "react-router-dom";
import UserProvider from "../../common/UserProvider";

it("renders without crashing", function () {
  render(
    <UserProvider>
      <CreateFolderForm />
    </UserProvider>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <CreateFolderForm />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
