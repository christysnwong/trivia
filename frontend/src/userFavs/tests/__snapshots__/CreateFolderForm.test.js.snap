// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card mb-5"
  >
    <div
      class="card-body"
    >
      <form>
        <div
          class="mb-2"
        >
          <label
            class="form-label"
            for="username"
          >
            Create a Folder: 
          </label>
          <input
            class="form-control"
            id="folderName"
            maxlength="20"
            name="folderName"
            placeholder="New Folder Name"
            value=""
          />
        </div>
        <button
          class="btn btn-secondary mt-4 me-3"
          method="get"
          type="button"
        >
          Cancel
        </button>
        <button
          class="btn btn-primary mt-4"
          method="post"
          type="submit"
        >
          Create
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
